infra/Pos/Communication/Types/Protocol.hs

```diff
@@ -204,13 +206,16 @@ type ListenersWithOut m = ([ListenerSpec m], OutSpecs)
 
 -- | Parser for PeerId. Any base64 string.
 peerIdParser :: P.Parser PeerId
-peerIdParser = fmap PeerId P.base64Url
+peerIdParser = do
+    bytes <- P.base64Url
+    when (BS.length bytes /= 14) $ Monad.fail "PeerId must be exactly 14 bytes"
```

Comment:
Do we really need `MonadFail` here? I know using exceptions here would be overkill, but people more knowledgeable than me (i.e. @snoyman in the Haskell best practices meetup) have advised against using this class at all, if possible.

@neongreen can you comment on this?